packages/theme/nuxt.config.js

```diff
@@ -3,200 +3,208 @@ import theme from './themeConfig';
 import webpack from 'webpack';
 import * as dotenv from 'dotenv';
 import urlJoin from 'url-join';
+import getStore from './getStore';
 
 dotenv.config({ path: `./../../.env.${process.env.NODE_ENV}` });
 
 const serverConfig = {
   port: process.env.PORT || 3000,
   host: process.env.HOST || '0.0.0.0',
-  baseUrl: process.env.BASE_URL || 'http://localhost:3000'
+  baseUrl: process.env.BASE_URL || 'http://localhost:3000',
+  backendUrl: process.env.BACKEND_URL
 };
 
-export default {
-  ssr: true,
-  server: {
-    port: serverConfig.port,
-    host: serverConfig.host
-  },
-  head: {
-    title: 'Vue Storefront',
-    meta: [
-      { charset: 'utf-8' },
-      { name: 'viewport', content: 'width=device-width, initial-scale=1' },
-      {
-        hid: 'description',
-        name: 'description',
-        content: process.env.npm_package_description || ''
-      }
-    ],
-    link: [
-      {
-        rel: 'icon',
-        type: 'image/x-icon',
-        href: '/favicon.ico'
-      },
-      {
-        rel: 'preconnect',
-        href: 'https://fonts.gstatic.com',
-        crossorigin: 'crossorigin'
-      },
-      {
-        rel: 'preload',
-        href: 'https://fonts.googleapis.com/css?family=Raleway:300,400,400i,500,600,700|Roboto:300,300i,400,400i,500,700&display=swap',
-        as: 'style'
-      },
-      {
-        rel: 'stylesheet',
-        href: 'https://fonts.googleapis.com/css?family=Raleway:300,400,400i,500,600,700|Roboto:300,300i,400,400i,500,700&display=swap',
-        media: 'print',
-        onload: 'this.media=\'all\'',
-        once: true
-      }
-    ]
-  },
-  loading: { color: '#fff' },
-  plugins: ['~/plugins/i18n.js'],
-  buildModules: [
-    // to core
-    '@nuxtjs/composition-api/module',
-    '@nuxt/typescript-build',
-    '@nuxtjs/google-fonts',
-    '@nuxtjs/pwa',
-    '@nuxtjs/style-resources',
-    ['@vue-storefront/spree/nuxt', {}],
-    ['@vue-storefront/nuxt', {
+export default async () => {
+  const store = await getStore({
+    backendUrl: serverConfig.backendUrl
```

Comment:
Any reason for not replacing nuxt.config.js with .ts extension? We're lacking type safety in many places here, eg. `const store: any`.

Reply:
I mean type inference works here anyway because of how typescript is configured in this project. But yes we should change that here and in all the other places where files are defined as `.js`. I'll make a separate PR for that.

Reply:
https://github.com/vuestorefront-community/spree/pull/301